gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/query_service/portal_page/PortalNavigationItemsQueryServiceImpl.java

```diff
@@ -0,0 +1,98 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.apim.infra.query_service.portal_page;
+
+import io.gravitee.apim.core.exception.TechnicalDomainException;
+import io.gravitee.apim.core.portal_page.model.PortalArea;
+import io.gravitee.apim.core.portal_page.model.PortalNavigationItem;
+import io.gravitee.apim.core.portal_page.model.PortalNavigationItemId;
+import io.gravitee.apim.core.portal_page.query_service.PortalNavigationItemsQueryService;
+import io.gravitee.apim.infra.adapter.PortalNavigationItemAdapter;
+import io.gravitee.repository.exceptions.TechnicalException;
+import io.gravitee.repository.management.api.PortalNavigationItemRepository;
+import java.util.Collection;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Service;
+
+@Service
+public class PortalNavigationItemsQueryServiceImpl implements PortalNavigationItemsQueryService {
+
+    private final PortalNavigationItemRepository portalNavigationItemRepository;
+    private final PortalNavigationItemAdapter portalNavigationItemAdapter = PortalNavigationItemAdapter.INSTANCE;
+    private static final Logger logger = LoggerFactory.getLogger(PortalNavigationItemsQueryServiceImpl.class);
+
+    public PortalNavigationItemsQueryServiceImpl(@Lazy final PortalNavigationItemRepository portalNavigationItemRepository) {
+        this.portalNavigationItemRepository = portalNavigationItemRepository;
+    }
+
+    @Override
+    public PortalNavigationItem findByIdAndEnvironmentId(String environmentId, PortalNavigationItemId id) {
+        try {
+            var result = portalNavigationItemRepository.findById(id.json());
+            if (result.isPresent() && result.get().getEnvironmentId().equals(environmentId)) {
+                return portalNavigationItemAdapter.toEntity(result.get());
+            }
+            return null;
+        } catch (TechnicalException e) {
+            String errorMessage = String.format(
+                "An error occurred while finding portal navigation item by id %s and environmentId %s",
+                id,
+                environmentId
+            );
+            logger.error(errorMessage, e);
+            throw new TechnicalDomainException(errorMessage, e);
+        }
+    }
+
+    @Override
+    public Collection<PortalNavigationItem> findByParentIdAndEnvironmentId(String environmentId, PortalNavigationItemId parentId) {
+        try {
+            var results = portalNavigationItemRepository.findAllByParentIdAndEnvironmentId(parentId.json(), environmentId);
+            return results.stream().map(portalNavigationItemAdapter::toEntity).collect(Collectors.toList());
```

Comment:
If you know that you are returning a List. I would simply put "List" instead of Collection as the returned type.